### PR TITLE
Adding a paragraph about updating multiple packages during an update

### PR DESCRIPTION
**Pull request by weaverryan**
| Q | A |
| --- | --- |
| Doc fix? | no |
| New docs? | no |
| Applies to | 2.3+ |
| Fixed tickets | n/a |

Hi guys!

For my 2.7 upgrade, I needed to run `composer update symfony/symfony twig/twig`. I'm trying to include a note to help other people. I'd appreciate opinions on wording :).

Thanks!

### CODE REVIEW

**Comment by weaverryan** [comment]
Note: Meant to be merged into 2.3

**Comment by WouterJ** [comment]
Isn't this quite strange? Composer update should trigger a partial update afaik, which means it should also update twig/twig if a newer version is required. Are we sure Symfony has the correct version constraint for twig?

**Comment by xabbuh** [comment]
Yeah, that sounds weird.

@weaverryan Do you remember what required you to use a newer Twig version than the one you had installed?

**Comment by javiereguiluz** [comment]
In my case I didn't have problems with Twig but with some Doctrine dependencies. I know that you won't agree but I believe that when you perform a "brutal update" (like updating Symfony version) you should always execute `composer update` instead of updating just a minor set of packages.

**Comment by weaverryan** [comment]
@xabbuh: yes, symfony/symfony required a newer version of twig/twig than I had installed. I have no twig/twig in *my* composer.json. So I also thought that Composer would be smart enough to update twig, even though it's not an arg in my update command, but that's not the case.

So what should we do? The error - and getting it working- is confusing. But the brutal update (with no args) really means people need to have great version constraints.

@javiereguiluz - was your error with doctrine/common? I'm wondering because maybe this is solved (or mostly solved) if you update symfony/symfony + whatever libs it depends on directly.

Thanks!

**Comment by WouterJ** [comment]
Maybe @Seldaek can give us some clues if this behaviour is the expected one (`composer update symfony/symfony` only updating symfony/symfony and *not* its dependencies) and we should document how to work around it or if this is actually a bug in Composer?

From a usage point of view, the current behaviour seems rather strange/hard to use.

**Comment by javiereguiluz** [comment]
@weaverryan in my case it was a problem with Doctrine fixtures ... but it only happened in one project. I'm afraid that there are so many different constraints configurations, that for a lot of developers doing `composer update symfony/symfony` won't work :(

**Comment by xabbuh** [comment]
@weaverryan @javiereguiluz I'm curious. Could you please show how your `composer.json`  file looked like before the update?

**Comment by Seldaek** [comment]
Composer used to be smart enough as you say and update dependencies as well when you did a partial update, but then for some packages you end up updating half the project, so it's not so great. Either you have to whitelist all the deps yourself or you can use `composer update --with-dependencies symfony/symfony` and then it'll include the deps automatically but it might end up updating more than you'd like it to.

**Comment by WouterJ** [comment]
Thanks for the information @Seldaek. So I think we should change the instructions to `composer update --with-dependencies symfony/symfony` as it seems to be at least a bit safer for people that don't have the knowledge to write good version constraints.

**Comment by weaverryan** [comment]
I just updated this with `--with-dependencies` - it has indeed helped a lot of people to use that for the 2.7 upgrade.

**Comment by xabbuh** [comment]
:+1:

File: cookbook/upgrade/_update_dep_errors.rst.inc [hunk 1]
```diff
@@ -0,0 +1,19 @@
+Dependency Errors
+~~~~~~~~~~~~~~~~~
+
+If you get a dependency error, it may simply mean that you need to upgrade
+other Symfony dependencies too. In that case, try the following command:
+
+    $ composer update symfony/symfony --with-dependencies
+
+This updates ``symfony/symfony`` and *all* packages that it depends on,
+will include several other packages. By using tight version constraints in
```
Comment (WouterJ) [inline cookbook/upgrade/_update_dep_errors.rst.inc]: The second part of the first sentences doesn't sound quite right. Maybe something like "which will include several other packages. [...]" is better?

**Pull request closed without merging**